File: 刷题/cpp/1828. 统计一个圆中点的数目.cpp
class Solution {
public:
    vector<int> countPoints(vector<vector<int>>& points, vector<vector<int>>& queries) {
        int n=queries.size(),m=points.size();
        vector<int> ans;
        for(int i=0;i<n;i++){
            int cnt=0;
            for(int j=0;j<m;j++){
                int cx=queries[i][0]-points[j][0],cy=queries[i][1]-points[j][1],r=queries[i][2];
                if(cx*cx+cy*cy<=r*r){
                    cnt++;
                }
            }
            ans.emplace_back(cnt);
        }
        return ans;
    }
};

File: 刷题/golang/1828. 统计一个圆中点的数目.go
func countPoints(points [][]int, queries [][]int) []int {
    n:=len(queries)
    ans:=make([]int,n)
    for i,x:=range queries{
        for _,y:=range points{
            cx:=x[0]-y[0]
            cy:=x[1]-y[1]
            r:=x[2]
            if cx*cx+cy*cy<=r*r{
                ans[i]++
            }
        }
    }
    return ans
}